Bump IMGUI to v1.73

SemMulder opened the pull request:
This PR "depends" on #56.

This bumps IMGUI to v1.73.

dertseha commented on imgui.go:
```diff
@@ -670,9 +670,9 @@ func TreePop() {
-	C.iggSetNextTreeNodeOpen(castBool(open), C.int(cond))
+// SetNextItemOpen sets the open/collapsed state of the following tree node.
+func SetNextItemOpen(open bool, cond Condition) {
+	C.iggSetNextItemOpen(castBool(open), C.int(cond))
```
Removing a function would require a major version bump, which I'd like to avoid if possible.
Recommended: Keep `SetNextTreeNodeOpen()` and comment it as "deprecated", which IDEs tend to point out. Have it then call the new, recommended function. This then is identical to the compatibility mode in imgui.h .

dertseha commented:
Skimming over the changes in `imgui.h`, there's at least one enum that has changed as well: that of the keyboard identifier. As such, the values of the `Key*` constants in `imgui.go` would be wrong.

I'll update the guideline in `CONTRIBUTING.md` about such cases - because we had this issue at an upgrade already. I might also add some tests to catch such things.

Do you want to do another diligent pass over the changes? I'll probably do a merge into a preparation branch and then work on details anyway. For instance, I see several changes in comments to functions in `imgui.h` which I guess would also be good to be ported to the Go documentation.

An upgrade is quite some work, thank you for starting the initiative!

SemMulder commented:
I just added a commit that updates all the `*Flags` according to their values in `imgui.h`. I replaced also usages of `iota` with explicit definition of the constants to prevent bugs when changing the constants.

dertseha commented on imgui.go:
```diff
@@ -982,16 +990,16 @@ func IsMouseDoubleClicked(button int) bool {
 
 // ColumnsV creates a column layout of the specified number of columns.
-func ColumnsV(count int, label string, flags int32) {
+func ColumnsV(count int, label string, border bool) {
```
This signature change (together with the deletion of the ColumnFlags) warrants a switch to v2.0.0 for the wrapper as it is an incompatible change.
It seems like this is inevitable. It may be a good reason for me to look into how to properly handle major releases with Go.

dertseha commented:
Thank you again for the modifications.
It seems there are more incompatible changes lurking than just the one removed method, so I guess I'll look into making a major release with that. This may take a few more days until I can get to it, probably over the holidays.

dertseha commented:
I'm taking this over now. Since v1.74 has been released meanwhile, with further incompatibilities, I'll even go one step further and upgrade to that as well.
